huaweicloud/services/rds/data_source_huaweicloud_rds_sqlserver_databases.go

```diff
@@ -0,0 +1,174 @@
+package rds
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk/pagination"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// @API RDS GET /v3/{project_id}/instances/{instance_id}/database/detail
+func DataSourceSQLServerDatabases() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: resourceSQLServerDatabasesRead,
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"instance_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"character_set": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"databases": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     sqlserverDatabasesSchema(),
+			},
+		},
+	}
+}
+
+func sqlserverDatabasesSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"character_set": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+	return &sc
+}
+
+func resourceSQLServerDatabasesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var (
+		listSQLServerDatabasesHttpUrl = "v3/{project_id}/instances/{instance_id}/database/detail?page=1&limit=100"
+		listSQLServerDatabasesProduct = "rds"
+	)
+	listSQLServerDatabasesClient, err := cfg.NewServiceClient(listSQLServerDatabasesProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating RDS client: %s", err)
+	}
+
+	listSQLServerDatabasesPath := listSQLServerDatabasesClient.Endpoint + listSQLServerDatabasesHttpUrl
+	listSQLServerDatabasesPath = strings.ReplaceAll(listSQLServerDatabasesPath, "{project_id}",
+		listSQLServerDatabasesClient.ProjectID)
+	listSQLServerDatabasesPath = strings.ReplaceAll(listSQLServerDatabasesPath, "{instance_id}",
+		fmt.Sprintf("%v", d.Get("instance_id")))
+
+	listSQLServerDatabasesResp, err := pagination.ListAllItems(
+		listSQLServerDatabasesClient,
+		"page",
+		listSQLServerDatabasesPath,
+		&pagination.QueryOpts{MarkerField: ""})
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving RDS SQLServer databases")
```

Comment:
CheckDeletedDiag should not be used in data_source

Reply:
done